dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -236,6 +238,38 @@ public Page<Event> getEvents(EventQueryParams queryParams, PageParams pageParams
     return new Page<>(events, pageParams, () -> getEventCount(queryParams));
   }
 
+  public List<Event> getEvents(
+      EventQueryParams queryParams, Map<String, Set<String>> psdesWithSkipSyncTrue) {
+    if (queryParams.isSynchronizationQuery()
+        && psdesWithSkipSyncTrue != null
```

Comment:
I am thinking if it is better to reuse the existing fetchEvents method.. but inside it check 
if (queryParams.isSynchronizationQuery()

If yes, then proceed to get psdesWithSkipSynctrue only then. and then resume. Does that sound reasonable?